thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java

```diff
@@ -190,4 +201,92 @@ public Response computeSeverity(@NotNull @QueryParam("collection") String collec
 
     return Response.ok(severity.toString(), MediaType.TEXT_PLAIN_TYPE).build();
   }
+
+  /**
+   * Breaks down the given range into consecutive monitoring windows as per function definition
+   * Regenerates anomalies for each window separately
+   *
+   * As the anomaly result regeneration is a heavy job, we move the function from Dashboard to worker
+   * @param id anomaly function id
+   * @param monitoringWindowStartTime The start time of the monitoring window (in milli-second)
+   * @param monitoringWindowEndTime The start time of the monitoring window (in milli-second)
+   * @param isForceBackfill false to resume backfill from the latest left off
+   * @return HTTP response of this request
+   * @throws Exception
+   */
+  //private final String detectionHost = "localhost";
+  //private final int detectionPort = 1867;
+  @POST
+  @Path("/{id}/regenerateAnomaliesInRange")
```

Comment:
There's a method few lines above in this file, called backfill. It is doing exactly what this method is doing. Please make sure we don't duplicate this functionality. Either remove the backfill method, or make this regenerate method call backfill method.